cmd/proplet/main.go

```diff
@@ -2,158 +2,121 @@ package main
+	service, err := proplet.NewService(ctx, cfg.ChannelID, cfg.ThingID, cfg.ThingKey, cfg.RegistryURL, cfg.RegistryToken, cfg.LivelinessInterval, mqttPubSub, logger, wazero)
 	if err != nil {
-		logger.Error("Error initializing service", slog.Any("error", err))
+		logger.Error("failed to initialize service", slog.Any("error", err))
 
-		return fmt.Errorf("service initialization error: %w", err)
+		return
 	}
 
 	if err := service.Run(ctx, logger); err != nil {
-		logger.Error("Error running service", slog.Any("error", err))
+		logger.Error("failed to run service", slog.Any("error", err))
 
-		return fmt.Errorf("service run error: %w", err)
+		return
 	}
 
-	return nil
-}
-
-func configureLogger(level string) *slog.Logger {
-	if err := logLevel.UnmarshalText([]byte(level)); err != nil {
-		log.Printf("Invalid log level: %s. Defaulting to info.\n", level)
-		logLevel = slog.LevelInfo
-	}
-
-	logHandler := slog.NewJSONHandler(os.Stdout, &slog.HandlerOptions{
-		Level: logLevel,
+	g.Go(func() error {
+		return server.StopSignalHandler(ctx, cancel, logger, svcName)
 	})
 
-	return slog.New(logHandler)
-}
-
-func loadWASMFile(path string, logger *slog.Logger) ([]byte, error) {
-	logger.Info("Loading WASM file", slog.String("path", path))
-	wasmBytes, err := os.ReadFile(path)
-	if err != nil {
-		return nil, fmt.Errorf("failed to read WASM file: %w", err)
+	if err := g.Wait(); err != nil {
+		logger.Error(fmt.Sprintf("%s service exited with error: %s", svcName, err))
 	}
-
-	return wasmBytes, nil
 }
 
-func checkRegistryConnectivity(registryURL string, logger *slog.Logger) error {
-	ctx, cancel := context.WithTimeout(context.Background(), registryTimeout)
+func checkRegistryConnectivity(ctx context.Context, registryURL string, registryTimeout time.Duration) error {
+	ctx, cancel := context.WithTimeout(ctx, registryTimeout)
 	defer cancel()
 
-	client := http.Client{}
-
-	logger.Info("Checking registry connectivity", slog.String("url", registryURL))
+	client := http.DefaultClient
 
 	req, err := http.NewRequestWithContext(ctx, http.MethodGet, registryURL, http.NoBody)
 	if err != nil {
-		logger.Error("Failed to create HTTP request", slog.String("url", registryURL), slog.Any("error", err))
-
 		return fmt.Errorf("failed to create HTTP request: %w", err)
 	}
 
 	resp, err := client.Do(req)
 	if err != nil {
-		logger.Error("Failed to connect to registry", slog.String("url", registryURL), slog.Any("error", err))
-
-		return fmt.Errorf("failed to connect to registry URL '%s': %w", registryURL, err)
+		return fmt.Errorf("failed to connect to registry URL: %w", err)
 	}
 	defer resp.Body.Close()
 
 	if resp.StatusCode != http.StatusOK {
-		logger.Error("Registry returned unexpected status", slog.String("url", registryURL), slog.Int("status_code", resp.StatusCode))
-
-		return fmt.Errorf("registry URL '%s' returned status: %s", registryURL, resp.Status)
+		return fmt.Errorf("fegistry returned unexpected status: %d", resp.StatusCode)
```

Comment:
return fmt.Errorf("registry returned unexpected status: %d", resp.StatusCode)
```suggestion
		return fmt.Errorf("fegistry returned unexpected status: %d", resp.StatusCode)
```